src/metrics.ts

```diff
@@ -138,6 +157,24 @@ export class Metrics {
   }
 
   trackHost(source: string, cache?: CacheItem): CacheItem|undefined {
+    if (this.#maxHostRatio) {
+      // only track if ratio limits enabled
+      this.#hostRatioCounts.set(source, (this.#hostRatioCounts.get(source) || 0) + 1);
+      this.#hostRatioCount++;
+
+      if (this.#hostRatioCount >= this.#hostRatioMaxCount) {
```

Comment:
Just to make sure I understand - so if I set the throttling at 20%, we will check every 200 requests whether we need to throttle, but if I set it to 10% we will check every 100 requests? Curious why the throttling percentage affects the history size in that way (unless I'm misunderstanding this code)...

Reply:
That's correct. The greater the ratio the greater the window needed for a "fair" sampling. Small enough window to be responsive, large enough to be a fair approximation.

Reply:
Interesting - if you have any resources on this type of sampling I'd be interested to read them - I don't have much experience with sampling algorithms